cleanlab/segmentation/filter.py

```diff
@@ -0,0 +1,180 @@
+      
+    n_jobs: int, optional
+      Number of processes for multiprocessing (default value = 1). Only used on Linux.
+      If `n_jobs=None`, will use either the number of: physical cores if psutil is installed, or logical cores otherwise.
+    
+    verbose : bool, optional
+      Whether to suppress print statements or not.
+
+    **kwargs:
+        scores_only: optional
+        Set to True to return a score for each image. Meant for internal call in 
+        ``cleanlab.semantic_segmentation.rank.get_label_quality_scores``
+
+    
+    Returns
+    -------
+    label_issues : np.ndarray
+      Returns a boolean **mask** for the entire dataset
+      where ``True`` represents a pixel label issue and ``False`` represents an example that is
+      accurately labeled with high confidence.
+
+    """
+    scores_only = kwargs.get("scores_only", False)
+    
+    def check_input(labels: np.ndarray, pred_probs: np.ndarray) -> None:
+        if len(labels.shape) != 3:
+            raise ValueError("labels must have a shape of (N, H, W)")
+
+        if len(pred_probs.shape) != 4:
+            raise ValueError("pred_probs must have a shape of (N, K, H, W)")
+
+        num_images, height, width = labels.shape
+        num_images_pred, num_classes, height_pred, width_pred = pred_probs.shape
+
+        if num_images != num_images_pred or height != height_pred or width != width_pred:
+            raise ValueError("labels and pred_probs must have matching dimensions for N, H, and W")
+        
+        #Check downsample
+        if height%downsample!=0 or width%downsample!=0:
+            raise ValueError(f"Height {height} and width {width} not divisible by downsample value of {downsample}")
+
+    
+    def downsample_arrays(labels: np.ndarray, pred_probs: np.ndarray, factor: int = 1) -> tuple[np.ndarray, np.ndarray]:
+        if factor == 1:
+            return labels, pred_probs
+        num_image, num_classes, h, w = pred_probs.shape
+        small_labels = np.round(labels.reshape((num_image, h // factor, factor,
+                                                w // factor, factor)).mean(4).mean(2))
+        small_pred_probs = pred_probs.reshape((num_image, num_classes, h // factor, factor,
+                                               w // factor, factor)).mean(5).mean(3)
+
+        # We want to make sure that pred_probs are renormalized
+        row_sums = small_pred_probs.sum(axis=1)
+        renorm_small_pred_probs = (small_pred_probs / np.expand_dims(row_sums, 1))
+
+        return small_labels, renorm_small_pred_probs
+    
+    def flatten_and_preprocess_masks(labels: np.ndarray, pred_probs: np.ndarray) -> tuple[np.ndarray, np.ndarray]:
+        _, num_classes, _, _ = pred_probs.shape
+        labels_flat = labels.flatten().astype(int)
+        pred_probs_flat = np.moveaxis(pred_probs, 0, 1).reshape(num_classes, -1)
+
+        return labels_flat, pred_probs_flat.T
+    
+    
+    ##
+    check_input(labels, pred_probs)
+    
+    
```

Comment:
What is this code for? If this is testing, please move it into `cleanlab/tests/test_segmentation.py. If this is example code then maybe into the main function's header or better documented?